backend/test/Test/UploadSpec.hs

```diff
@@ -101,6 +107,19 @@ spec = withContext $ startWithInitial $ do
                   , mrIsEnabled = False
                   }
         onlyDisabled secondaryMeasurements `shouldBe` secondaryDisabled
+
+    it "rollbacks targets and compounds due processing error" $ \ctx -> do
+      runRIO ctx (uploadFile' (SqlId 1) (SqlId 1) "" (error "e") "" sampleFile)
+        `shouldThrow` anyErrorCall
+
+      targets <- runRIO ctx $ Library.getTargets Nothing Nothing Nothing
+      compounds <- runRIO ctx $ Library.getCompounds Nothing Nothing Nothing
+      experiments <- runRIO ctx $ Dashboard.getExperiments Nothing Nothing Nothing
+
+      -- We do not use 0 here, because we add some data in upper tests
+      length targets `shouldBe` 4
+      length compounds `shouldBe` 5
```

Comment:
Also currently these checks don't test the actual problem, because even if you add `sampleFile` successfully, you won't add new compounds/targets because they are already added.

Reply:
agree